portfolio/src/main/webapp/viewpuzzle.html

```diff
@@ -3,6 +3,7 @@
     <meta charset="UTF-8">
     <title>Solve</title>
     <script src="script.js"></script>
+    <script type="text/javascript" src="snapfit.js"></script>
```

Comment:
You'll need to copy the contents of snapfit.js into a file of the same name and include it in our code.

The source code for it can be found here:
http://www.netzgesta.de/snapfit/snapfit.js

Reply:
Also swap this with the line above. Our own script.js depends on snapfit.js so snapfit.js must be loaded  first:
```suggestion
<script src="script.js"></script>
<script type="text/javascript" src="snapfit.js"></script>
```